### DISCUSSION TITLE
Auto Amp power control

### DISCUSSION THREAD

**Discussion by kevin88252** [dtcooper/raspotify#441] [Ideas]
First of all, thank you for this super easy and great software :-)

i use it to listen on Spotify on my older Harman Kardon Amp....

one great feature would be if it could run a custom script when a user is connected and another if disconnected, so it could send a IR-Signal to turn my amp on and set it to the right input source and off after i stopped listening to music....

is there already a way to do something like this?

**Comment by JasonLG1979** [comment]
> is there already a way to do something like this?

The short answer is no if you expect to just hook a USB IR blaster and have it do what you're asking.

The long answer is yes but you gotta roll your own:

You can run a script on events with the `--onevent PATH_TO_SCRIPT` where `PATH_TO_SCRIPT` is the absolute path to the event script you want to run. You would then use `--emit-sink-events` to send sink events. ***Sink events are blocking*** And as far as the audio backends are concerned there is no such thing as pause, and manual track changes close and then reopen the sink. (normal track changes do not unless the `--disable-gapless` flag is passed.)

In your case the blocking aspect of sink events actually works to your advantage as I'm sure it takes a sec or 2 for your receiver to actually turn on. In your script I would make all situations you want to ignore the sink events basically noops and send your IR-Signal asynchronously and put in a sleep for how many secs it usually takes your receiver to kick on go to the right input.

Here are the relevant events (non sink events are non-blocking):
https://github.com/librespot-org/librespot/blob/dev/src/player_event_handler.rs

**Reply by JasonLG1979** [reply]
See: https://github.com/dtcooper/raspotify/discussions/455#discussioncomment-1665604
